cou/apps/subordinate.py

```diff
@@ -0,0 +1,80 @@
+#  Copyright 2023 Canonical Limited
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+"""Subordinate application class."""
+import logging
+
+from cou.apps.app import SUBORDINATES, AppFactory, OpenStackApplication
+from cou.exceptions import ApplicationError
+from cou.steps import UpgradeStep
+from cou.utils.openstack import OpenStackRelease
+
+logger = logging.getLogger(__name__)
+
+
+@AppFactory.register_application(SUBORDINATES)
+class OpenStackSubordinateApplication(OpenStackApplication):
+    """Subordinate application class."""
+
+    def __post_init__(self) -> None:
+        """Initialize the Application dataclass."""
+        self.channel = self.status.charm_channel
```

Comment:
There are `channel` getter/setter methods created later in the class, why do we still need a `self.channel` variable here?

Reply:
We are validating `channel` and for that you can see that the setter accepts a value named `charm_channel`. That is why is still needed. This [article](https://www.programiz.com/python-programming/property) has an example of that.

Reply:
I need to revert this change because when creating object channel is set to "" and it creates exception since we do not allow this.

Reply:
I think using getter/setter is more appropriate. Maybe if we remove `channel` as a required parameter to instantiate `OpenStackApplication` and do it on `__post_init__` is a better solution. I believe the same behavior that you were using on ` OpenStackSubordinateApplication` should be applied on `OpenStackApplication`. Changing on `OpenStackApplication` would fix for both.

Reply:
done